tests/UnitTests/Foundation/NSIndexPathTests.mm

```diff
@@ -0,0 +1,167 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include <TestFramework.h>
+#import <Foundation/Foundation.h>
+#import <Starboard/SmartTypes.h>
+
+TEST(NSIndexPath, Init) {
+    NSIndexPath* path = [[[NSIndexPath alloc] init] autorelease];
+    ASSERT_NE(nil, path);
+    EXPECT_EQ(0, [path length]);
+    EXPECT_EQ(NSNotFound, [path indexAtPosition:0]);
+}
+
+TEST(NSIndexPath, InitWithIndex) {
+    NSIndexPath* path = [[[NSIndexPath alloc] initWithIndex:5] autorelease];
+    ASSERT_NE(nil, path);
+    EXPECT_EQ(1, [path length]);
+    EXPECT_EQ(5, [path indexAtPosition:0]);
+    EXPECT_EQ(NSNotFound, [path indexAtPosition:1]);
+}
+
+TEST(NSIndexPath, InitWithIndexes) {
+    NSUInteger indexes[3] = { 1, 2, 3 };
+    NSIndexPath* path = [[[NSIndexPath alloc] initWithIndexes:indexes length:3] autorelease];
+    ASSERT_NE(nil, path);
+    EXPECT_EQ(3, [path length]);
+    EXPECT_EQ(1, [path indexAtPosition:0]);
+    EXPECT_EQ(2, [path indexAtPosition:1]);
+    EXPECT_EQ(3, [path indexAtPosition:2]);
+    EXPECT_EQ(NSNotFound, [path indexAtPosition:4]);
+}
+
+// These are disabled because the methods are not yet supported on the reference testing platform
+OSX_DISABLED_TEST(NSIndexPath, IndexPathForRowInSection) {
+    NSIndexPath* path = [NSIndexPath indexPathForRow:2 inSection:1];
+    ASSERT_NE(nil, path);
+    EXPECT_EQ(2, [path length]);
+    EXPECT_EQ(1, [path indexAtPosition:0]);
+    EXPECT_EQ(2, [path indexAtPosition:1]);
+    EXPECT_EQ(NSNotFound, [path indexAtPosition:2]);
+}
+
+OSX_DISABLED_TEST(NSIndexPath, IndexPathForItemInSection) {
```

Comment:
hmm is this the same case?
If it's not supported (why?) add the comment here too.